XenModel/Actions/Pool/PoolAction.cs

```diff
@@ -86,6 +86,10 @@ protected static bool FixCpus(Pool pool, List<Host> hostsToCpuMask, Func<HostAbs
             // Mask the CPUs, and reboot the hosts (simultaneously, as they must all be on separate connections)
             foreach (Host host in hostsToCpuMask)
             {
+                // No CPU masking is needed for Dundee or greater hosts 
+                if (Helpers.DundeeOrGreater(host))

```

Comment:
I think we should Assert here, rather than (or perhaps, as well as) just continuing, because Dundee hosts shouldn't get here.